src/core/openthread-core-default-config.h

```diff
@@ -282,6 +282,16 @@
 #endif
 
 /**
+ * @def OPENTHREAD_CONFIG_SRV_ALOCS_PER_DEVICE
+ *
+ * The maximum number of supported Service ALOCs registrations per device.
```

Comment:
nit: "per device" implies to me that this configuration applies to the entire network rather than just the local device.  How about "for this device" in the comment instead and `OPENTHREAD_CONFIG_MAX_SERVER_ALOCS`.